lib/reek/cli/options.rb

```diff
@@ -213,6 +214,15 @@ def set_utility_options
           puts parser
           exit
         end
+        parser.on_tail('-l', '--list', 'List all available smell detectors') do
+          puts 'All available smell detectors:\n'
+          DetectorRepository.smell_types.map do |full_name|
+            full_name.to_s.split('::').last
```

Comment:
Don't we have a method on `SmellDetector` that gives the short name already?

Reply:
Good catch! And we do [indeed](https://github.com/troessner/reek/blob/master/lib/reek/smell_detectors/base_detector.rb#L94) ;)